server/src/Masna3/Server/Error.hs

```diff
@@ -1,12 +1,54 @@
 module Masna3.Server.Error where
 
 import Data.Aeson
+import GHC.Generics (Generic)
+import Masna3.Api.File.FileId
 import Servant (ServerError (..), err500)
 
 data Masna3Error
   = TooManyRows Text
+  | FileNotFound FileId
+  | InvalidTransition InvalidTransitionError
   deriving stock (Eq, Ord, Show)
 
+data InvalidTransitionError
+  = NotPendingToUploaded FileId
+  deriving stock (Eq, Generic, Ord, Show)
+  deriving anyclass (ToJSON)
+
 toServerError :: Masna3Error -> ServerError
 toServerError = \case
-  TooManyRows t -> err500{errBody = encode t}
+  TooManyRows t ->
+    err500
+      { errBody =
+          encode
+            ErrorResponse
+              { errorType = "TooManyRows"
+              , context = Just $ object ["ownerId" .= t]
+              }
+      }
+  FileNotFound t ->
+    err500
+      { errBody =
+          encode
+            ErrorResponse
+              { errorType = "FileNotFound"
+              , context = Just $ object ["fileId" .= t]
+              }
+      }
+  InvalidTransition (NotPendingToUploaded t) ->
+    err500
+      { errBody =
+          encode
+            ErrorResponse
+              { errorType = "InvalidTransition NotPendingToUploaded"
+              , context = Just $ object ["fileId" .= t]
+              }
+      }
+
+data ErrorResponse = ErrorResponse
+  { errorType :: Text
+  , context :: Maybe Value
+  }
+  deriving stock (Generic)
+  deriving anyclass (ToJSON)
```

Comment:
No need for that. :) 

The intuition is good, but we actually want to have a `context` field that can hold more data than we can actually send back – for security & privacy purposes.

My suggestion is that we can put the metadata of `context` as logging metadata, not in the response. This is done with the [`localData`](https://hackage-content.haskell.org/package/log-base-0.12.1.0/docs/Log-Class.html#v:localData) of our logging framework. 

For instance:

```diff
- Error.throwError (FileNotFound fileId)
+ Log.localData ["file_id" .= fileId] $
+   Error.throwError (FileNotFound fileId)
```